pkg/collector/listeners/docker.go

```diff
@@ -351,14 +359,21 @@ func (s *DockerService) GetHosts() (map[string]string, error) {
 	ips := make(map[string]string)
 	du, err := docker.GetDockerUtil()
 	if err != nil {
-		return ips, err
+		return nil, err
 	}
 	cInspect, err := du.Inspect(string(s.ID), false)
 	if err != nil {
 		return nil, fmt.Errorf("failed to inspect container %s", string(s.ID)[:12])
 	}
 	for net, settings := range cInspect.NetworkSettings.Networks {
-		ips[net] = settings.IPAddress
+		if len(settings.IPAddress) > 0 {
```

Comment:
Is there's a use case for this if or is for safety?

Reply:
rancher injects a `none` network with no IP, we don't want to pick it up